packages/@lwc/engine-core/src/framework/rendering.ts

```diff
@@ -753,14 +753,17 @@ function allocateInSlot(vm: VM, children: VNodes, owner: VM) {
             continue;
         }
 
-        let slotName = '';
+        let slotName: string | number | boolean = '';
         if (isVBaseElement(vnode)) {
-            slotName = (vnode.data.attrs?.slot as string) ?? '';
+            slotName = vnode.data.attrs?.slot ?? '';
         } else if (isVScopedSlotFragment(vnode)) {
             slotName = vnode.slotName;
         }
 
-        const vnodes: VNodes = (cmpSlotsMapping[slotName] = cmpSlotsMapping[slotName] || []);
+        const normalizedSlotName = slotName.toString();
```

Comment:
If the `slotName` is bound to an object created like `Object.create(null)`, this toString will throw.